cmd/snap/cmd_run.go

```diff
@@ -1186,7 +1186,13 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	// https://www.freedesktop.org/wiki/Software/systemd/ControlGroupInterface/
 	_, appName := snap.SplitSnapApp(snapApp)
 	needsTracking := true
-	if app := info.Apps[appName]; hook == "" && app != nil && app.IsService() {
+
+	// Do not track apps that don't wish to be tracked.
+	if app := info.Apps[appName]; hook == "" && app.Untracked {
```

Comment:
The original code also checks `.. app != nil .. `, so we probably need it here too. A separate question is whether that check is maybe unnecessary by the time we reach here?

Reply:
Safety first :)

Reply:
This part is entirely gone now.